docs/specifications/url.rst

```diff
@@ -0,0 +1,112 @@
+URLs
+====
+
+The goal of this document is to completely specify the construction and use of the URLs by Tahoe-LAFS for service location.
+This includes, but is not limited to, the original Foolscap-based URLs.
+These are not to be confused with the URI-like capabilities Tahoe-LAFS uses to refer to stored data.
+An attempt is also made to outline the rationale for certain choices about these URLs.
+The intended audience for this document is Tahoe-LAFS maintainers and other developers interested in interoperating with Tahoe-LAFS or these URLs.
+
+Background
+----------
+
+Tahoe-LAFS first used Foolscap_ for network communication.
+Foolscap connection setup takes as an input a Foolscap URL or a *fURL*.
+A fURL includes three components:
+
+* the base32-encoded SHA1 hash of the DER form of an x509v3 certificate
+* zero or more network addresses
```

Comment:
dang, really? you don't have to have a least one?

Reply:
At least according to the Foolscap-supplied parser you don't.  I don't know if this is *intended* or not ... it's just allowed.

Maybe I'll add a footnote explaining where I got this conclusion.